### PR TITLE
[runtime] Add stack marks to a few icalls. NFC.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by vargaz** [comment]
@monojenkins commit apidiff

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins commit apidiff

**Comment by vargaz** [comment]
@monojenkins build failed

File: mono/metadata/reflection-internals.h [hunk 1]
```diff
@@ -137,4 +137,7 @@ mono_class_from_mono_type_handle (MonoReflectionTypeHandle h);
 MonoMethod*
 mono_runtime_get_caller_no_system_or_reflection (void);
```
Comment (marek-safar) [inline mono/metadata/reflection-internals.h]: Would be nice to remove this unused code now

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by akoeplinger** [comment]
Merging since the api-snapshot was bumped and otherwise we'd have conflicts with other PRs.

**Pull request merged**